### PR TITLE
fixed an issue where env support for edgeapplication was not comprehe…

### PR DESCRIPTION
**Pull request by tangming1996**
…nsive

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
The differentiated configuration of edgeapplication env supports only the key-value format. Errors will be reported in other formats
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by WillardHu** [comment]
Please make the required items check successful.

File: cloud/pkg/controllermanager/edgeapplication/overridemanager/envoverrider.go [hunk 1]
```diff
@@ -216,14 +217,87 @@ func convertToEnvVar(value interface{}) (*corev1.EnvVar, error) {
 
 func convertToEnvVarSource(value map[string]interface{}) (corev1.EnvVarSource, error) {
```
Comment (WillardHu) [inline cloud/pkg/controllermanager/edgeapplication/overridemanager/envoverrider.go:218]: You can obtain a value using the NestedXxx function in the package k8s.io/apimachinery/pkg/apis/meta/v1/unstructured（helpers.go）

**Review by Shelley-BaoYue — COMMENTED** [review]
/lgtm

**Comment by WillardHu** [comment]
Please cherry-pick to release-1.16

**Pull request merged**